### DISCUSSION TITLE
How does the template in output filename work?

### DISCUSSION THREAD

**Discussion by axi92** [yao-pkg/pkg#75] [Q&A] [answered]
The --help says "output file name or template for several files" but how does that work?

I like to have the architecture always appended to the filename.
The reason is that I use docker buildx build to build in a multi-platform container simultaneously 2 different archs. PKG only sees one arch inside the container and does not append the arch.
When I copy those files from the container they have the same name.

Is there as simple way to use the template for the names?

**Comment by robertsLando** [comment]
I generally put the arch in the name myself when building single executables, that's quite easy if you use `--output` to specify the output filename

**Reply by axi92** [reply]
I know but since I  build with a generic dockerfile and use the --platform param for docker buildx build I don't know inside the container what arch I am building.

Thats the prompt who I call the Dockerfile:
`docker buildx build --build-arg PKG_TARGET=node20-linux --platform linux/amd64,linux/arm64/v8 -f ./Dockerfile --output type=local,dest=./ .`
I only use it to build the binary with pkg and then copy it from there to the local disc.

I was trying to figure out how too use the template but I was lost in the source :grinning: 

``` Dockerfile
FROM node:20 AS builder

ARG PKG_TARGET

COPY . /root
WORKDIR /root
RUN npm install
RUN npx pkg -t ${PKG_TARGET} app/service.js -c package.json -o appname

FROM scratch AS export
COPY --from=builder /root/appname /
```

**Reply by robertsLando** [reply]
>  I don't know inside the container what arch I am building.

```
RUN npx pkg -t ${PKG_TARGET} app/service.js -c package.json -o appname-${arch}
```

**Reply by axi92** [reply, marked as answer]
Where does the value come from? `${arch}` is empty.
But I solved it that way:
`RUN arch=$(dpkg --print-architecture); npx pkg -t node20-linux app/service.js -c package.json -o appname-${arch}`

**Reply by robertsLando** [reply]
@axi92 `arch` it's a command that prints the architecture: https://ioflood.com/blog/arch-command-in-linux/#:~:text=TL%3BDR%3A%20What%20is%20the,typing%20arch%20in%20your%20terminal.&text=In%20this%20example%2C%20we've,command%20in%20a%20Linux%20terminal.

Glad you fixed it anyway :)
